### PR TITLE
Volcano plot

### PR DESCRIPTION
**Pull request by federicomarini**
Starting the PR to facilitate comments & code review 😉 

A couple of things might still be needed:

- [x] unit tests set up for this particular function: we can simply start inspired by the examples, and then cover the possible edge cases as we know what behaviors to expect.
- [x] name of the function - shall we keep this or switch to anything like `signature_volcano`
- [x] adding some element in the app tour to make sure this feature gets highlighted (we'd need an anchor - can be obtained from inspecting the source of the web app itself - and a simple text description
- [x] updating the news, and bumping the version number + add you as author in the package DESCRIPTION
- [x] some small descriptive section in the vignette itself (ideally, close to where the `gs_heatmap` is)

### CODE REVIEW

File: R/ggs_volcano.R [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    if (!all(genelist %in% rownames(res_de))) {
+      not_there <- genelist[!(genelist %in% rownames(res_de))]
+      warning("Some of the provided gene ids were not found in the SummarizedExperiment",
+              "\nNot found: ", not_there)
+    }
+    thisset_members_ids <- intersect(rownames(res_de), genelist)
+    thisset_name <- "Custom list"
+  }
+
+  thisset_members_ids
```
Comment (federicomarini) [inline R/ggs_volcano.R:117]: I think this one can safely be removed - I also saw it in the signature heatmap function, it is probably a remnant of debugging sessions

File: R/ggs_volcano.R [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+  padj_complete <- sapply(padj_complete, function(x) -log10(x))
+  
+  log2FoldChange_complete <- res_de[complete_genes_ids, "log2FoldChange"]
+  
+  gene_set_belong <- complete_genes_ids %in% thisset_members_ids
+  filter_info_complete <- filter_info_complete & gene_set_belong
+
+  
+  thisset_complete_data <- data.frame(complete_genes_ids, padj_complete, log2FoldChange_complete, filter_info_complete, gene_set_belong)
+  colnames(thisset_complete_data) <- c("genes", "logTransformedpvalue", "log2FoldChange", "significant", "belonging")
```
Comment (federicomarini) [inline R/ggs_volcano.R:134]: feel free to indent a la 
```
data.frame(
  complete_genes_ids,
  padj_complete,
  ...
)
```

It is better to avoid too long lines - it is a harmless thing, but sometimes it helps to also "give you an idea" of the structure.

File: R/ggs_volcano.R [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+            color = "p-value <= FDR") + 
+       scale_x_continuous(limits = limit_x) +
+       scale_color_manual(labels = c("significant", "not significant"), 
+                          breaks = c("TRUE", "FALSE"),
+                          values = c(color, "grey25")) +
+       scale_alpha_manual(breaks = c("TRUE", "FALSE"),
+                          values = c(1, 1/10)) +
+       theme_bw() +
+       theme(legend.title = element_text(size = 11, face = "bold"),
+             legend.text = element_text(size = 10))
```
Comment (federicomarini) [inline R/ggs_volcano.R:166]: this one can be indented like the one below (I am an avid user of Cmd + I on Mac 😉 )

File: R/ggs_volcano.R [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+#'            
+#' alternatively
+#'
+#' # genelist <- sapply(anno_df$gene_name, function(x) str_contains(x, "CD"))
+#' 
+#' # ggs_volcano(res_de,
+#' #             res_enrich,
+#' #             anno_df,
+#' #             genelist = genelist
+#' #             )
```
Comment (federicomarini) [inline R/ggs_volcano.R:77]: I'd make the call here more explicit, no problem in "showing that we just take a small bunch of genes on ourselves".
Plus: let the code be run, it would be an extra second or so, but no need to worry for that - and if we pass the genes directly, we avoid having to use the `stringr` package 😬 
You could just call the list `cdgenes_list` as in here, or you could use the `chemokines_list`, whatever suits you best (just trying to avoid naming the variable as the parameter itself)

File: R/ggs_volcano.R [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+#' res_enrich <- shake_topGOtableResult(topgoDE_macrophage_IFNg_vs_naive)
+#' res_enrich <- get_aggrscores(res_enrich, res_de, anno_df)
+#'
+#' ggs_volcano(res_de,
+#'            res_enrich,
+#'            anno_df,
+#'            geneset_id = res_enrich$gs_id[1]
+#'            )
+#'            
+#' alternatively
```
Comment (federicomarini) [inline R/ggs_volcano.R:69]: I'm puzzled that this line alone did not trigger any error (there is no real comment before that) 😄

**Comment by federicomarini** [comment]
Looking quite good, well done.
All set once we address the points in the initial message.

**Comment by AnnekathrinSilvia** [comment]
Changed branch according to comments, merge updated branch in new pull request
